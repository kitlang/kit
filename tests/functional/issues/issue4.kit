trait MyTrait {
    public function echo(): Void;
}

implement MyTrait for Int {
    public function echo() {
        printf("hello from Int %i\n", this);
    }
}

function main() {
    var a: Int = 10;
    // This will create a temp variable which relies on A, so temp variable
    // generation must be in the correct order or the generated code won't
    // compile.
    (((a + 5) as Int) as Box[MyTrait]).echo();
}
