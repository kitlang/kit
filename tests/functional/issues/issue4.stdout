hello from Int 15
